### ISSUE TITLE
[Package Request] ioBroker/Node.js >= 8.12

### ISSUE THREAD

**Issue by florianbehrens** [SynoCommunity/spksrc#3682]
Name: ioBroker
Description: IoT/home automation platform
Website: http://www.iobroker.net/
Software documentation: c
Build/Installation documentation: https://www.npmjs.com/package/iobroker
Source code: https://www.npmjs.com/package/iobroker
License: MIT

ioBroker is available via `npm` but requires a more recent Node.js version than provided by Synology. So this package request is merely a request for a newer Node.js package. I'd offer to step in and provide one, is there a chance you'd accept a PR?

**Comment by chickenandpork** [comment]
@florianbehrens PRs are accepted and welcome: there can sometimes be delays between submission and official builds.

It's easier if you've got others willing to confirm your beta SPKs on other architectures than your own (the ~25% of C/C++ in node.js seems to be in the shipping interpreter).

Out of curiosity, is there a static test that would allow a non-nodejs person like myself effectively run a cut-n-paste touch-test that confirms it's OK?  If I was to receive your updated node.js, is there a one-liner that checks that it's sane and functional ?

**Comment by florianbehrens** [comment]
Ok, thanks for your quick reply.

Regarding smoke test: I'm not a Node.js expert, too. But I'd suggest to install your favorite node tool using `npm` and give it a try. `npm` runs on Node, too, so that gives some first feedback.

E.g.,
````
npm i -g nasa-cli
nasa -t
````

**Comment by ymartin59** [comment]
Hello. I have not investigated specifically ioBroker BUT many NPM packages embeds C/C++ sources that would require a compiler to be available on Synology DSM system.
The proper solution is to cross-compile dependencies and embed them in SynoCommunity packages like already done with Python modules. Refer to #2793 for details.

So your PR proposal to create a Node.js v10 will only be able to install pure JavaScript  NPM packages.
By the way, that is the same for Synology official v8 package... Just a restriction to be aware of before invoking "nmp -i".

Another tip to be aware of, global installation will probably fails as requiring root privileges and because Synology packages have their own private install location. Running as a service also requires a dedicated non-privileged user.

As a partial conclusion: many constraints to meet before expecting to run an NodeJS application on Synology.

**Comment by florianbehrens** [comment]
@ymartin59

Yes, ioBroker only runs with `--no-optional` on targets w/o compiler. Some features (serialport and UDP discovery seem not be available then).

Another Problem is that ioBroker's Installation process is fairly complicated and simple npm Installation is not officially supported. But I managed to get it running on my DS216j. I still have to figure out how to fit that in an SPK.

**Comment by hgy59** [comment]
@florianbehrens still trying to build a ioBroker package?
Synology now offers Node.js up to version 22 (on DSM 7.2). Therefore, ioBroker should be feasible.
